test/e2e/storage/csi_defs.go

```diff
@@ -197,3 +199,59 @@ func csiHostPathPod(
 	framework.ExpectNoError(framework.WaitForPodRunningInNamespace(client, ret))
 	return ret
 }
+
+func csiGCEPDSetup(
+	client clientset.Interface,
+	config framework.VolumeTestConfig,
+	teardown bool,
+	f *framework.Framework,
+	nodeSA *v1.ServiceAccount,
+	controllerSA *v1.ServiceAccount,
+) {
+	// Get API Objects from manifests
+	nodeds, err := manifest.DaemonSetFromManifest("test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml", config.Namespace)
+	framework.ExpectNoError(err, "Failed to create DaemonSet from manifest")
+	nodeds.Spec.Template.Spec.ServiceAccountName = nodeSA.GetName()
+
+	controllerss, err := manifest.StatefulSetFromManifest("test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml", config.Namespace)
+	framework.ExpectNoError(err, "Failed to create StatefulSet from manifest")
+	controllerss.Spec.Template.Spec.ServiceAccountName = controllerSA.GetName()
+
+	controllerservice, err := manifest.SvcFromManifest("test/e2e/testing-manifests/storage-csi/gce-pd/controller_service.yaml")
+	framework.ExpectNoError(err, "Failed to create Service from manifest")
+
+	// Got all objects from manifests now try to delete objects
+	err = client.CoreV1().Services(config.Namespace).Delete(controllerservice.GetName(), nil)
```

Comment:
Each test case should already run in a different namespace. Do you really need to delete these at the beginning?

Reply:
Not strictly necessary -- defensive programming though? lmk if you think it should be removed